plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/PrometheusServer.java

```diff
@@ -20,24 +20,40 @@
 import java.net.URI;
 import java.time.Duration;
 
+import static org.testcontainers.utility.MountableFile.forClasspathResource;
 import static org.testng.Assert.fail;
 
 public class PrometheusServer
         implements Closeable
 {
     private static final int PROMETHEUS_PORT = 9090;
-    private static final String PROMETHEUS_DOCKER_IMAGE = "prom/prometheus:v2.15.1";
+    private static final String DEFAULT_VERSION = "v2.15.1";
+    public static final String LATEST_VERSION = "v2.35.0";
     private static final Integer MAX_TRIES = 120;
     private static final Integer TIME_BETWEEN_TRIES_MILLIS = 1000;
 
+    public static final String USER = "admin";
+    public static final String PASSWORD = "password";
+
     private final GenericContainer<?> dockerContainer;
 
     public PrometheusServer()
     {
-        this.dockerContainer = new GenericContainer<>(PROMETHEUS_DOCKER_IMAGE)
+        this(DEFAULT_VERSION);
+    }
+
+    public PrometheusServer(String version)
+    {
+        this.dockerContainer = new GenericContainer<>("prom/prometheus:" + version)
                 .withExposedPorts(PROMETHEUS_PORT)
                 .waitingFor(Wait.forHttp("/"))
                 .withStartupTimeout(Duration.ofSeconds(120));
+        if (version.equals(LATEST_VERSION)) {
```

Comment:
The basic auth is introduced only in the latest version i.e 2.35 ?

Reply:
It was introduced in `v2.24.0`.